File: spec/Linear_Search_Spec.rb
require 'spec_helper'

describe 'Linear_Search' do

  before :each do
    @container = My_Container.new [2,34,5,6,8,12,3,5,5,6,3]
    @search = Linear_Search.new @container
  end

  describe 'find' do
    it 'should return the indexs of the value given' do
      @search.find(5).should eq([2,7,8])
      @search.find(3).should eq([6,10])
    end

    it 'should return an empty array if no matches found' do
      @search.find(1).should eq([])
    end
  end
end

File: spec/javascripts/LinearSearchSpec.js
describe('LinearSearch', function() {

    describe('search', function() {
        var ls, container;

        beforeEach(function() {
            ls = new LinearSearch();
            container = new MyContainer(10);
        });

        it('should return the first index found containing the value given', function() {
            expect(ls.search(container, 5)).toEqual(5);
            expect(ls.search(container, 9)).toEqual(9);
            expect(ls.search(container, 1)).toEqual(1);
        });

        it('should return undefined if the value search does not exist', function() {
            expect(ls.search(container, 10)).toEqual(undefined);
        });
    });
});
